servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -163,11 +164,11 @@ private void cleanupOnWriteTerminated() {
 
     private DefaultNettyConnection(
             Channel channel, ExecutionContext<?> executionContext,
-            CloseHandler closeHandler, FlushStrategy flushStrategy,
+            CloseHandler closeHandler, FlushStrategy flushStrategy, final boolean offloadAsyncClose,
             @Nullable Long idleTimeoutMs, Protocol protocol, @Nullable SSLSession sslSession,
             @Nullable ChannelConfig parentChannelConfig, DataObserver dataObserver, boolean isClient,
             UnaryOperator<Throwable> enrichProtocolError) {
-        super(channel, executionContext.executor());
+        super(channel, offloadAsyncClose ? executionContext.executor() : immediate());
```

Comment:
What prevents the usage of `executionContext.executionStrategy()` and it's to-be-added new method: `asyncCloseOffloaded()`? I suppose that would simplify things greatly. Especially having in mind that `AsyncCloseable` is a core concept in `concurrent-api`.